app/javascript/react/components/molecules/Calendar/Calendar.tsx

```diff
@@ -1,27 +1,165 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useSelector } from "react-redux";
+import moment, { Moment } from "moment";
 
-import { selectLatestThreeMonthsDailyAverages } from "../../../store/fixedStreamSelectors";
+import { selectThreeMonthsDailyAverage } from "../../../store/movingStreamSelectors";
 import { Month } from "./atoms/Month";
-import * as S from "./Calendar.style";
 import { useTranslation } from "react-i18next";
 import HeaderToggle from "./HeaderToggle/HeaderToggle";
+import { useAppDispatch } from "../../../store/hooks";
+import {
+  movingData,
+  fetchNewMovingStream,
+} from "../../../store/movingCalendarStreamSlice";
+import { ScrollCalendarButton } from "./atoms/ScrollCalendarButton/ScrollCalendarButton";
+import * as S from "./Calendar.style";
+
+interface MovableCalendarData {
+  zeroDate: string;
+  currentData: string;
+  direction: number | undefined;
+  triggerDirectionUpdate: number;
+}
 
 const Calendar = () => {
-  const threeMonthsData = useSelector(selectLatestThreeMonthsDailyAverages);
+  const SEEN_MONTHS_NUMBER = 3;
+
+  const threeMonthsData = useSelector(selectThreeMonthsDailyAverage);
+  const movingCalendarData = useSelector(movingData);
   const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const [dateReference, setDateReference] = useState<MovableCalendarData>({
+    zeroDate: "",
+    currentData: "",
+    direction: undefined,
+    triggerDirectionUpdate: 0,
+  });
+
+  const updateMoveValue = (direction: 1 | -1) => {
+    setDateReference((prevState) => {
+      return {
+        ...prevState,
+        direction: direction,
+        triggerDirectionUpdate: prevState.triggerDirectionUpdate + 1,
+      };
+    });
+  };
+
+  const handleBackwardMovement = (dateMoment: moment.Moment) => {
+    const newEndMoment = dateMoment.date(1).subtract(1, "months");
+    setIsButtonDisabled(false);
+    return newEndMoment;
+  };
+
+  const handleForwardMovement = (dateMoment: moment.Moment) => {
```

Comment:
What do you think about using just one function for handling movement?

`const handleMovement = (dateMoment: moment.Moment, direction: number) => {
    const newEndMoment = dateMoment.clone().date(1).add(direction, "months");
    const zeroDateMoment = moment(dateReference.zeroDate, "YYYY-MM-DD");
    const shouldDisableButton = newEndMoment.isSameOrAfter(zeroDateMoment);
    setIsButtonDisabled(shouldDisableButton);
    return shouldDisableButton ? null : newEndMoment;
  };`

Reply:
I find it clearer when I can read it separately for -> and <- so if there is no pressure I would like it to stay as there are some differences